packages/core/src/common/utils/replace-observable-object.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { runInAction } from "mobx";
+
+export function replaceObservableObject<Key extends string | number | symbol>(target: Partial<Record<Key, unknown>>, source: Partial<Record<Key, unknown>>): void {
```

Comment:
I don't think should ever be used. The pattern that it introduces is troublesome. This will replace existing reference of an property with another. By looking at the use cases and the name of function it seems that it's meant to be used with observables, so it would mean that we replace reference of an observable with another. I don't see any good reason why we should do so, since we could just update the value. This opens up a whole new category of bugs where computing observable doesn't do it's job at all because the observable which it's dereferencing is replaced with another observable. I'd prefer not knowing if something like this works properly in MobX.